machine_test.go

```diff
@@ -270,6 +270,7 @@ func (s *machineSuite) TestCreateDevice(c *gc.C) {
 	})
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(device.InterfaceSet()[0].Name(), gc.Equals, "eth4")
+	c.Assert(device.InterfaceSet()[0].VLAN().ID(), gc.Equals, subnet.VLAN().ID())
```

Comment:
I tried to chase it down to check, but are you sure the subnet vlan id was different from the interface vlan id to start with? Did this fail before the fix?